client/app/models/event.coffee

```diff
@@ -101,6 +101,81 @@ module.exports = class Event extends ScheduleItem
     getDefaultColor: -> '#008AF6'
 
 
+    # A shared event has at least one attendees with a cozy invitation
+    isShared: ->
+        attendees = @get 'attendees'
+        cozyAttendees = attendees?.find (attendee) ->
+            return attendee.shareWithCozy
+        return cozyAttendees?
+
+
+    # Try to get a shareID (called typically just after saving),
+    tryGetShareID: (numtries, delay, callback) ->
+
+        @fetch
+            success: (model, response) =>
+                shareID = model.get 'shareID'
+                if shareID
+                    callback null, shareID
+                    return
+
+                triesLeft = --numtries;
+
+                if numtries
+                    setTimeout =>
+                        @tryGetShareID triesLeft, delay, callback
+                    , delay
+                else
+                    callback 'Could not retrieve shareID, maximum \
+                        number of tries exceeded',
+                        null
+            error: (model, response) ->
+                callback(response.error or response, null)
+
+
+    # Fetch the sharing object to handle any error
+    onShareIDChange: ->
+        @fetchSharing (err, sharing) =>
+            if err
+                throw
+                    name: 'EventSharingError',
+                    event: @
+                    message: 'cannot retrieve sharing : ' + err
```

Comment:
@frankrousseau 
My idea here was to throw an error with a basic message and translate the later as possible, if possible just before printing it to the user. But maybe there is a problem with the concatenation.
What do you think I should do ? Pass the the previous `err` message in another property like `prevousError` or `stack` ?

And sould I translate the error message as soon as possible ? (In my opinion could cause log parsing issue, same error having different internal messages, but I am open-minded on this point :) )

(Also, by dealing with those asynchronous errors, I am trying to prepare a better error handling in the whole client application).